### PR TITLE
whalebird: 4.7.4 -> 5.0.7

### PR DESCRIPTION
**Pull request by Weathercold**
###### Description of changes

changelog: <https://github.com/h3poteto/whalebird-desktop/releases>

notable changes:
- Add hide-to-tray on start option (3942)
- Re-implement layout and sidebar (4037)
- Show reply target message on compose (4098)
- Show character limit in compose (4099)
- Show quote target message on compose (4100)
- Add validation for compose (4121)
- Auto-complete in compose (4139)
- Add shortcut key for post (4269)
- **Update electron version to 21.4.4** (4304)
- **Change license to GPL** (4363)
- Add scroll bar if there are many hashtags or lists (4421)
- Add shortuct for reloading with CMD+R (4422)
- Tons of dependency bumps and fixes

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/whalebird/default.nix [hunk 1]
```diff
@@ -83,8 +83,8 @@ stdenv.mkDerivation rec {
     description = "Electron based Mastodon, Pleroma and Misskey client for Windows, Mac and Linux";
     homepage = "https://whalebird.social";
     sourceProvenance = with sourceTypes; [ binaryNativeCode ];
-    license = licenses.mit;
-    maintainers = with maintainers; [ wolfangaukang colinsane ];
+    license = licenses.gpl3;
```
Comment (uninsane) [inline pkgs/applications/misc/whalebird/default.nix:86]:
```suggestion
    license = licenses.gpl3Only;
```
`gpl3` is [deprecated](https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206). and looking upstream, they seem to have licensed it under "GPL 3.0" without any mention of "or any later version".

**Review by uninsane — CHANGES_REQUESTED** [review]
built and tested basic functionality on x86_64: looks good. trying to build it for aarch64 but it fails:

```console
$ NIXPKGS_ALLOW_INSECURE=1 nix build --impure '.#pkgsCross.aarch64-multiplatform.whalebird' --keep-failed
...
$ nix log /nix/store/3d6sd7yflvb7vpzddv6c8l2f94z5d776-whalebird-aarch64-unknown-linux-gnu-5.0.6.drv
warning: The interpretation of store paths arguments ending in `.drv` recently changed. If this command is now failing try again with '/nix/store/3d6sd7yflvb7vpzddv6c8l2f94z5d776-whalebird-aarch64-unknown-linux-gnu-5.0.6.drv^*'
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
building
node:internal/fs/utils:347
    throw err;
    ^

Error: ENOENT: no such file or directory, open '/build/opt/Whalebird/resources/app.asar.unpacked/node_modules/better-sqlite3/build/node_gyp_bins/python3'
    at Object.openSync (node:fs:601:3)
    at Object.readFileSync (node:fs:469:35)
    at module.exports.readFileSync (/nix/store/3lzirw2j0brpr7j5r62n8vwjs4dn68ml-asar-3.2.0/lib/node_modules/asar/lib/disk.js:110:17)
    at module.exports.extractAll (/nix/store/3lzirw2j0brpr7j5r62n8vwjs4dn68ml-asar-3.2.0/lib/node_modules/asar/lib/asar.js:204:28)
    at Command.<anonymous> (/nix/store/3lzirw2j0brpr7j5r62n8vwjs4dn68ml-asar-3.2.0/lib/node_modules/asar/bin/asar.js:72:10)
    at Command.listener [as _actionHandler] (/nix/store/3lzirw2j0brpr7j5r62n8vwjs4dn68ml-asar-3.2.0/lib/node_modules/asar/node_modules/commander/index.js:413:31)
    at Command._parseCommand (/nix/store/3lzirw2j0brpr7j5r62n8vwjs4dn68ml-asar-3.2.0/lib/node_modules/asar/node_modules/commander/index.js:914:14)
    at Command._dispatchSubcommand (/nix/store/3lzirw2j0brpr7j5r62n8vwjs4dn68ml-asar-3.2.0/lib/node_modules/asar/node_modules/commander/index.js:865:18)
    at Command._parseCommand (/nix/store/3lzirw2j0brpr7j5r62n8vwjs4dn68ml-asar-3.2.0/lib/node_modules/asar/node_modules/commander/index.js:882:12)
    at Command.parse (/nix/store/3lzirw2j0brpr7j5r62n8vwjs4dn68ml-asar-3.2.0/lib/node_modules/asar/node_modules/commander/index.js:717:10) {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: '/build/opt/Whalebird/resources/app.asar.unpacked/node_modules/better-sqlite3/build/node_gyp_bins/python3'
}

Node.js v18.16.0
```

you think that's an issue on my machine due to the `--impure` flag (can't build it without that `NIXPKGS_ALLOW_INSECURE` flag since electron 21 is marked EOL), or does this break aarch64 for others too?

**Comment by Weathercold** [comment]
> you think that's an issue on my machine due to the `--impure` flag (can't build it without that `NIXPKGS_ALLOW_INSECURE` flag since electron 21 is marked EOL), or does this break aarch64 for others too?

I don’t think the error is related to `--impure` flag. As the error happened in the build phase, my guess is that `asar` can’t find `node_modules/better-sqlite3/build/node_gyp_bins/python3` in `app.asar`’s file index. Probably an issue with upstream’s build?

Unfortunately I’m busy right now and can’t investigate in the following week.

File: pkgs/applications/misc/whalebird/default.nix [hunk 2]
```diff
@@ -3,20 +3,20 @@
...
 
   src = let
-    downloads = "https://github.com/h3poteto/whalebird-desktop/releases/download/${version}";
+    downloads = "https://github.com/h3poteto/whalebird-desktop/releases/download/v${version}";
   in
     if stdenv.system == "x86_64-linux" then
       fetchurl {
         url = downloads + "/Whalebird-${version}-linux-x64.tar.bz2";
-        sha256 = "sha256-jRtlnKlrh6If9wy3FqVBtctQO3rZJRwceUWAPmieT4A=";
+        sha256 = "sha256-cE8INLRwUoYTRuWu6lQNSB3DEq23Zo9zAa3eOkh2LJM=";
```
Comment (drupol) [inline pkgs/applications/misc/whalebird/default.nix:14]:
```suggestion
        hash = "sha256-cE8INLRwUoYTRuWu6lQNSB3DEq23Zo9zAa3eOkh2LJM=";
```

File: pkgs/applications/misc/whalebird/default.nix [hunk 3]
```diff
@@ -3,20 +3,20 @@
...
       }
     else if stdenv.system == "aarch64-linux" then
       fetchurl {
         url = downloads + "/Whalebird-${version}-linux-arm64.tar.bz2";
-        sha256 = "sha256-gWCBH2zfhJdJ3XUAxvZ0+gBHye5uYCUgX1BDEoaruxY=";
+        sha256 = "sha256-RtFUVWQIgcqqYR35600Xg3XMONLPjUdC2jY2aCc8+cY=";
```
Comment (drupol) [inline pkgs/applications/misc/whalebird/default.nix:19]:
```suggestion
        hash = "sha256-RtFUVWQIgcqqYR35600Xg3XMONLPjUdC2jY2aCc8+cY=";
```

**Review by drupol — COMMENTED** [review]
Hi,

I added a non-blocking comment, feel free to implement it if you want, if not, just let me know and I'll merge this immediately.

**Comment by Weathercold** [comment]
The aarch64-linux build should be fixed now.

**Review by drupol — APPROVED** [review]
Tested locally without any problem!

`NIXPKGS_ALLOW_INSECURE=1 nix run github:Weathercold/nixpkgs/whalebird-5.0.6#whalebird --impure`

**Pull request merged**